orderbook/src/fee.rs

```diff
@@ -90,6 +90,22 @@ impl UnsubsidizedFee {
     fn amount_in_sell_token(&self) -> f64 {
         self.gas_amount * self.gas_price * self.sell_token_price
     }
+
+    fn apply_fee_factor(
+        &self,
+        fee_configuration: &FeeSubsidyConfiguration,
+        app_data: AppId,
+    ) -> U256 {
+        let fee_in_eth = self.gas_amount * self.gas_price;
+        let discounted_fee_in_eth = (0f64).max(fee_in_eth - fee_configuration.fee_discount);
```

Comment:
Since this is incredibly unexpected, do you think its worth writing a warning log?